nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/AccessResource.java

```diff
@@ -851,6 +1086,47 @@ private void forwardToMessagePage(final HttpServletRequest httpServletRequest, f
         uiContext.getRequestDispatcher("/WEB-INF/pages/message-page.jsp").forward(httpServletRequest, httpServletResponse);
     }
 
+    private String determineLogoutMethod(String oidcDiscoveryUrl) {
+        Matcher accessTokenMatcher = REVOKE_ACCESS_TOKEN_LOGOUT_FORMAT.matcher(oidcDiscoveryUrl);
+        Matcher idTokenMatcher = ID_TOKEN_LOGOUT_FORMAT.matcher(oidcDiscoveryUrl);
+
+        if (accessTokenMatcher.find()) {
+            return REVOKE_ACCESS_TOKEN_LOGOUT;
+        } else if (idTokenMatcher.find()) {
+            return ID_TOKEN_LOGOUT;
+        } else {
+            return STANDARD_LOGOUT;
+        }
+    }
+
+    private URI oidcRequestAuthorizationCode(@Context HttpServletResponse httpServletResponse, String callback) {
+
+        final String oidcRequestIdentifier = UUID.randomUUID().toString();
+
+        // generate a cookie to associate this login sequence
+        final Cookie cookie = new Cookie(OIDC_REQUEST_IDENTIFIER, oidcRequestIdentifier);
```

Comment:
This cookie generation is done in two places (lines 1104 -1115 and 194-205). You could extract this to a private cookie generation method, something along the lines of:

httpServletResponse.addCookie(generateCookie(oidcRequestIdentifier));

Reply:
Is the cookie generation required for OIDC providers? Usually we use a token in the URL or HTTP header to associate requests rather than cookies.

Reply:
Yes, from what I can tell the cookie is being used to maintain state for the IDP login callback. Not exactly sure why this is a cookie. It's possible the IDP strips headers for the callback redirect. Not sure if @mtien-apache saw the JWT being stripped by the provider for callbacks? We could check with mcgilman why it was set up to use a cookie.

Reply:
At this time, I will continue using the cookie to maintain state and let the browser automatically associate it with the original request.